core/server/worker/src/main/java/alluxio/worker/netty/DataServerBlockReadHandler.java

```diff
@@ -61,9 +65,9 @@
     BlockReadRequestInternal(Protocol.ReadRequest request) throws Exception {
       super(request.getId(), request.getOffset(), request.getOffset() + request.getLength(),
           request.getPacketSize());
-      mBlockReader = mWorker
-          .readBlockRemote(request.getSessionId(), request.getId(), request.getLockId());
-      mWorker.accessBlock(request.getSessionId(), mId);
+      mLockId = mWorker.lockBlock(mSessionId, mId);
```

Comment:
This might not be sufficient because the current code (FileInStream#getBlockInStream) relies on the fact that once the block in stream is open,  the block is locked. Otherwise an exception will be thrown and we fall back to UFS. 

*** 
I was thinking to combine DataServerBlockReadHandler.java and DataServerUfsBlockReadHandler.java so that we do not need to hold the lock upon block in stream creation. Feel free to leave this to my PR (and bring the lock logic back) if you prefer.